src/pages/Portfolio/MyKiva/MyKivaPage.vue

```diff
@@ -176,6 +179,16 @@ const handleShowNavigation = () => {
 const handleBadgeSectionClicked = badge => {
 	state.value = STATE_JOURNEY;
 	selectedBadgeData.value = badge;
+	isEarnedSectionModal.value = false;
+	showBadgeModal.value = true;
+};
+
+const handleEarnedBadgeClicked = badge => {
+	const selectedTier = badge.achievementData?.tiers?.find(tierEl => tierEl.level === badge.level) ?? null;
+	state.value = STATE_EARNED;
+	tier.value = selectedTier;
+	selectedBadgeData.value = badge;
+	isEarnedSectionModal.value = true;
```

Comment:
Do we need this flag or could we just use `state`?

Reply:
We need it as, for Journey case we are not closing the modal for Earned badges, we are going back to Journey.

Reply:
Ah yup, good call.